src/main/java/org/dataone/bookkeeper/resources/UsagesResource.java

```diff
@@ -0,0 +1,342 @@
+                return usage;
+            }
+
+            // Ensure the caller is asssociated with the quota subject
+            // The quota subject is not in the usages table, it must be retrieved from the quota table.
+            Integer quotaId = usage.getQuotaId();
+            Quota quota = quotaStore.getQuota(quotaId);
+            Set<String> subjects = new HashSet<String>();
+            subjects.add(quota.getSubject());
+
+            Set<String> associatedSubjects =
+                this.dataoneAuthHelper.getAssociatedSubjects(caller, subjects);
+
+            if ( associatedSubjects.size() > 0 ) {
+                return usage;
+            } else {
+                throw new WebApplicationException(caller.getSubject() + " is not associated with this usage.", Response.Status.FORBIDDEN);
+            }
+        } catch (Exception e) {
+            String message = "The requested usage could not be retrieved: " + e.getMessage();
+            throw new WebApplicationException(message, Response.Status.EXPECTATION_FAILED);
+        }
+    }
+
+    /**
+     * Update the given usage, adding or update the usage row in the
+     * usages table.  Requires administrative authorization.
+     * @param context  the security context of the authenticated user
+     * @param usageId the usage identifier
+     * @return usage the usage object with the updated usage
+     * @throws WebApplicationException  if adjusting the usage fails
+     */
+    @Timed
+    @PUT
+    @PermitAll
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("{usageId}")
+    public Usage update(
+            @Context SecurityContext context,
+            @NotNull @Valid Usage usage,
+            @PathParam("usageId") @NotNull @Positive Integer usageId
+    ) throws WebApplicationException {
+        Usage updatedUsage;
+
+        // The calling user injected in the security context via authentication
+        Customer caller = (Customer) context.getUserPrincipal();
+        boolean isAdmin = this.dataoneAuthHelper.isAdmin(caller.getSubject());
+
+        if ( isAdmin ) {
+            try {
+                updatedUsage = usageStore.update(usage);
+            } catch (Exception e) {
+                String message = "Couldn't update the usage: " + e.getMessage();
+                throw new WebApplicationException(message, Response.Status.EXPECTATION_FAILED);
+            }
+        } else {
+            throw new WebApplicationException("Admin privilege is required to update a usage, " + caller.getSubject() + " is not authorized.", Response.Status.FORBIDDEN);
+        }
+        return updatedUsage;
+    }
+
+    /**
+     * Delete the usage
+     * @param usageId the usage id
+     * @return response 200 if deleted
+     * @throws WebApplicationException  a web app exception
+     */
+    @Timed
+    @DELETE
+    @RolesAllowed("CN=urn:node:CN,DC=dataone,DC=org")
```

Comment:
Again, we'll eventually remove this `@RolesAllowed` since permitted MNs will need to be able to delete usages with administrative object `MN.delete()` calls.